### ISSUE TITLE
If item is not cached, and get request is made which returns many items, find doesn't consider params and instead returns first item.

### ISSUE THREAD

**Issue by facultymatt** [FacultyCreative/ngActiveResource#6]
Consider the following case: 
1. User accesses 'Sensors.find(1)'
2. No sensors are cached so ngActiveResource makes a `GET` request to findURL which is `getSensorList.do?id=1`.
3. The API returns (because it's not smart) an array of sensors (all the sensors). 
4. The first item is returned to the user. This is NOT the item with `id === 1` because the API didn't recognize that the query params and incorrectly returned the complete list.

In the above case, YES the API should be a bit smarter. However ngActiveResource could be smarter itself and intervene between step 3 and 4:

 3a. `find()` attempts to match from the **new cache** that was created by the GET request.  

Basically, by doing one more check after a GET request which generates a new cache we could provider a more consistent interface.

**Comment by facultymatt** [comment]
As we discussed this is an unnecessary performance hit to compensate for poor APIs
